yt_dlp/networking/_urllib.py

```diff
@@ -212,7 +212,7 @@ def http_response(self, req, resp):
             resp.msg = old_resp.msg
             del resp.headers['Content-encoding']
         # brotli
-        if resp.headers.get('Content-encoding', '') == 'br':
+        if resp.headers.get('Content-encoding', '') == 'br' and brotli is not None:
```

Comment:
Isn't this worth an Exception? Though unless the sent `Accept-Encoding` is stripped of any `br`, a user option might have overridden the attempt to send only `SUPPORTED_ENCODINGS`.

See https://github.com/yt-dlp/yt-dlp/issues/5855#issuecomment-1363539612.